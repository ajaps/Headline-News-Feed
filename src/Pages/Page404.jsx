import React from 'react';
import PropTypes from 'prop-types';

import Footer from '../component/Footer.jsx';


/**
 * Represents a component 404 page.
 */
export default class page404 extends React.Component {

/**
   * return component that displays 404 page
   * @return {ReactElement} Markup
   */
  render() {
    return (
      <div>
        <h1>PAGE NOT FOUND </h1>
        <h3> You are reaching this page because the requested page cannot be found</h3>
        <h3> or it has been moved</h3>
        <h5><a href="/">Go to Home Page</a></h5>
        <Footer />
      </div>
    );
  }
}
