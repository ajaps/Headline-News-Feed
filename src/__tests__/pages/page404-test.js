import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Page404 from '../../pages/Page404.jsx';

describe('404 page', () => {
  let app;
  beforeEach(() => {
    app = shallow(<Page404 />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should contian', () => {
    expect(app.find('Page not found')).toBeTruthy();
  });
});
