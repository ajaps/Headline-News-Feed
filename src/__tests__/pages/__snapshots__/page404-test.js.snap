// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page should render as expected 1`] = `
<div>
  <h1>
    PAGE NOT FOUND 
  </h1>
  <h3>
     You are reaching this page because the requested page cannot be found
  </h3>
  <h3>
     or it has been moved
  </h3>
  <h5>
    <a
      href="/"
    >
      Go to Home Page
    </a>
  </h5>
  <Footer />
</div>
`;
